aptos-move/move-examples/token_objects/knight/sources/knight.move

```diff
@@ -0,0 +1,535 @@
+/// This module implements the knight token (non-fungible token) and the food tokens (fungible token).
+/// This module has the function to feed a knight token with food tokens to increase the knight's health point.
+module token_objects::knight {
+    use std::error;
+    use std::option;
+    use std::signer;
+    use std::string::{Self, String};
+    use aptos_std::string_utils;
+    use aptos_framework::event;
+    use aptos_framework::fungible_asset::{Self, Metadata};
+    use aptos_framework::object::{Self, Object};
+    use aptos_framework::primary_fungible_store;
+    use aptos_token_objects::collection;
+    use aptos_token_objects::property_map;
+    use aptos_token_objects::token;
+    use std::bcs;
+
+    /// The token does not exist
+    const ETOKEN_DOES_NOT_EXIST: u64 = 1;
+    /// The provided signer is not the creator
+    const ENOT_CREATOR: u64 = 2;
+    /// Attempted to mutate an immutable field
+    const EFIELD_NOT_MUTABLE: u64 = 3;
+    /// Attempted to burn a non-burnable token
+    const ETOKEN_NOT_BURNABLE: u64 = 4;
+    /// Attempted to mutate a property map that is not mutable
+    const EPROPERTIES_NOT_MUTABLE: u64 = 5;
+    // The collection does not exist
+    const ECOLLECTION_DOES_NOT_EXIST: u64 = 6;
+
+    /// The food collection name
+    const FOOD_COLLECTION_NAME: vector<u8> = b"Food Collection Name";
+    /// The food collection description
+    const FOOD_COLLECTION_DESCRIPTION: vector<u8> = b"Food Collection Description";
+    /// The food collection URI
+    const FOOD_COLLECTION_URI: vector<u8> = b"Food Collection URI";
```

Comment:
Maybe to avoid confusion you should make this look like a URL

Reply:
Done :) Thanks!